hw/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,13 @@
+import './Button.css'
+
+type ButtonProps = {
+    children: React.ReactNode;
+    onClick: (e: React.MouseEvent) => void;
+    disabled?: any
+}
+
+const Button: React.FC<ButtonProps> = ({children, onClick, disabled}) => {
+    return <button className='SearchButton' onClick = {!disabled ? onClick : () => console.log('not clicked')}>{children}</button>
```

Comment:
есть специальный атрибут `disabled` в `button`, так что ненужно писать логику, которую вы писали в `onClick`
```suggestion
    return <button className='SearchButton' disabled={disabled} onClick = {onClick}>{children}</button>
```